build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -235,6 +249,27 @@ service ActionCache {
 // `instance_name` is the instance name (see above), and `hash` and `size` are
 // the [Digest][build.bazel.remote.execution.v2.Digest] of the blob.
 //
+// Clients can download compressed data with a `resource_name` of the form
+// `{instance_name}/compressed-blobs/{compressor}/{uncompressed_hash}/{uncompressed_size}`
+// where compressor is the lowercase string form of a `Compressor.Value` enum
+// other than `identity` which is advertised by the server in
+// [CacheCapabilities.supported_compressor][build.bazel.remote.execution.v2.CacheCapabilities.supported_compressor],
+// `uncompressed_hash` and `uncompressed_size` represent the
+// [Digest][build.bazel.remote.execution.v2.Digest] of the downloaded blob
+// once decompressed. Clients SHOULD verify that the `uncompressed_hash` and
+// `uncompressed_size` match that of the downloaded data once uncompressed, and
+// take appropriate steps in the case of failure such as retrying a limited
+// number of times or surfacing an error to the user. Servers MUST provide
```

Comment:
We should probably make this a bit more general: 'servers MUST provide both compressed and uncompressed data for any blobs which is advertised as being available, regardless of how it was uploaded' or something like that.

Reply:
Added a statement to this effect.